sycl/source/detail/scheduler/commands.cpp

```diff
@@ -3185,6 +3187,20 @@ pi_int32 ExecCGCommand::enqueueImpQueue() {
 
     return PI_SUCCESS;
   }
+  case CG::CGTYPE::ProfilingTag: {
+    const PluginPtr &Plugin = MQueue->getPlugin();
+    // If the queue is not in-order, we need to insert a barrier. This barrier
+    // does not need output events as it will implicitly enforce the following
+    // enqueue is blocked until it finishes.
+    if (!MQueue->isInOrder())
+      Plugin->call<PiApiKind::piEnqueueEventsWaitWithBarrier>(
+          MQueue->getHandleRef(), 0, nullptr, nullptr);
+
+    Plugin->call<PiApiKind::piEnqueueTimestampRecordingExp>(
+        MQueue->getHandleRef(), false, 0, nullptr, Event);
```

Comment:
nit: There are a few instances where it would be nice to have inline comments naming literal parameters so it's clear what they are since they are unnamed e.g. `foo(/*foo*/nullptr, /*bar*/0)`

Reply:
Comments have been added.